tools/site_cobble/rdl_pkg/models.py

```diff
@@ -111,27 +111,67 @@ def __init__(self, **kwargs):
         super().__init__(**kwargs)
 
     @property
-    def used_bits(self):
+    def used_bits(self) -> int:
         return sum([x.width for x in self.packed_fields])
 
     @property
-    def packed_fields(self):
+    def packed_fields(self) -> List["Field"]:
         """
         Returns all the defined register fields, skipping any ReservedFields (undefined spaces)
         """
         return [x for x in self.fields if not isinstance(x, ReservedField)]
 
     @property
-    def encoded_fields(self):
+    def encoded_fields(self) -> List["Field"]:
         """
         Returns any register fields that have encodings
         """
         return [x for x in self.packed_fields if x.has_encode()]
+    
+    @property
+    def has_encoded_fields(self) -> bool:
+        """
+        True if any fields have an encoding, meaning we'll generate enumerated types for them
+        """
+        return len(self.encoded_fields) > 0
+    
+    @property
+    def reset_is_all_1s(self) -> bool:
+        """
+        True if all defined, non-reserved fields have a reset value of all 1s
+        """
+        for field in self.fields:
+            rst_prop = field.get_property("reset")
+            if (not isinstance(field, ReservedField) and 
+                (rst_prop is not None) and 
+                rst_prop != field.reset_1s):
+                    return False
+        return True
+    
+    @property
+    def reset_is_all_0s(self) -> bool:
+        """
+        True if all defined, non-reserved fields have a reset value of all 0s
+        """
+        return self.elaborated_reset == 0
```

Comment:
It jumps out to me how `reset_is_all_1s` looks so different than `reset_is_all_0s` when they essentially answer the same thing. If I'm understanding correctly, the 1s just looks field by field to check for 1s, while `elaborated_reset` will hand back the reset for the whole register. We can rely on `elaborated_reset` for the 0s case because we define `ReservedFields` to be 0 always. Am I following correctly?

Reply:
Yes, the ReservedFields don't contribute to the reset value which is otherwise shifted and bit-wise OR'd in so any '1's in the defined bits make this a non-0 but if it's 0, everything defined was zero.

The reset 1s is more complicated b/c we have to actively avoid looking at the ReservedField contribution since they aren't showing as 1s.